scripts/envDFL.sh

```diff
@@ -0,0 +1,243 @@
+        reset-environment                       reset all environment variables set by DynaFlow Launcher
+"
+}
+
+help() {
+    define_options
+    echo "$DYNAFLOW_LAUNCHER_USAGE"
+}
+
+set_commit_hook() {
+    $HERE/set_commit_hook.sh
+}
+
+set_environment() {
+    # dynawo vars
+    export DYNAWO_INSTALL_DIR=$DYNAWO_HOME
+    export LD_LIBRARY_PATH=$DYNAWO_INSTALL_DIR/lib:$LD_LIBRARY_PATH
+    export IIDM_XML_XSD_PATH=$DYNAWO_INSTALL_DIR/share/iidm/xsd
+    export DYNAWO_RESOURCES_DIR=$DYNAWO_INSTALL_DIR/share:$DYNAWO_INSTALL_DIR/share/xsd
+    export DYNAWO_DDB_DIR=$DYNAWO_INSTALL_DIR/ddb
+    export DYNAWO_IIDM_EXTENSION=$DYNAWO_INSTALL_DIR/lib/libdynawo_DataInterfaceIIDMExtension.so
+    export DYNAWO_LIBIIDM_EXTENSIONS=$DYNAWO_INSTALL_DIR/lib
+
+    # build
+    export DYNAFLOW_LAUNCHER_BUILD_DIR=$DYNAFLOW_LAUNCHER_HOME/buildLinux
+    export DYNAFLOW_LAUNCHER_INSTALL_DIR=$DYNAFLOW_LAUNCHER_HOME/installLinux
+    export_var_env DYNAFLOW_LAUNCHER_SHARED_LIB=OFF # same default value as cmakelist
+    export_var_env DYNAFLOW_LAUNCHER_USE_DOXYGEN=ON # same default value as cmakelist
+    export_var_env DYNAFLOW_LAUNCHER_BUILD_TESTS=ON # same default value as cmakelist
+    export_var_env DYNAFLOW_LAUNCHER_CMAKE_GENERATOR="Unix Makefiles"
+    export_var_env DYNAFLOW_LAUNCHER_PROCESSORS_USED=1
+
+    # Run
+    export DYNAFLOW_LAUNCHER_INSTALL=$DYNAFLOW_LAUNCHER_INSTALL_DIR
+    export LD_LIBRARY_PATH=$DYNAFLOW_LAUNCHER_HOME/lib64:$LD_LIBRARY_PATH
+    export_var_env DYNAFLOW_LAUNCHER_LOG_LEVEL=INFO # INFO by default
+
+    # other
+    set_commit_hook
+}
+
+reset_environment_variables() {
+    for var in $(printenv | grep DYNAFLOW_LAUNCHER_ | cut -d '=' -f 1); do
+        unset $var
+    done
+
+    ld_library_path_remove $DYNAFLOW_LAUNCHER_HOME/lib64
+    ld_library_path_remove $DYNAFLOW_LAUNCHER_HOME/lib
+
+    unset DYNAWO_HOME
+    unset DYNAWO_INSTALL_DIR
+    unset DYNAWO_RESOURCES_DIR
+    unset DYNAWO_DDB_DIR
+    unset DYNAWO_IIDM_EXTENSION
+    unset DYNAWO_LIBIIDM_EXTENSIONS
+    unset IIDM_XML_XSD_PATH
+}
+
+clean() {
+    rm -rf $DYNAFLOW_LAUNCHER_BUILD_DIR
+    rm -rf $DYNAFLOW_LAUNCHER_INSTALL_DIR
+}
+
+cmake_configure() {
+    cmake \
+        -S $DYNAFLOW_LAUNCHER_HOME \
+        -B $DYNAFLOW_LAUNCHER_BUILD_DIR \
+        -G $DYNAFLOW_LAUNCHER_CMAKE_GENERATOR \
+        -DCMAKE_BUILD_TYPE:STRING=$DYNAFLOW_LAUNCHER_BUILD_TYPE \
+        -DCMAKE_INSTALL_PREFIX:STRING=$DYNAFLOW_LAUNCHER_INSTALL_DIR \
+        -DDYNAWO_HOME:STRING=$DYNAWO_HOME \
+        -DBOOST_ROOT:STRING=$DYNAWO_HOME \
+        -DDYNAFLOW_LAUNCHER_LOCALE:STRING=$DYNAFLOW_LAUNCHER_LOCALE \
+        -DDYNAFLOW_LAUNCHER_SHARED_LIB:BOOL=$DYNAFLOW_LAUNCHER_SHARED_LIB \
+        -DDYNAFLOW_LAUNCHER_USE_DOXYGEN:BOOL=$DYNAFLOW_LAUNCHER_USE_DOXYGEN \
+        -DDYNAFLOW_LAUNCHER_BUILD_TESTS:BOOL=$DYNAFLOW_LAUNCHER_BUILD_TESTS
+}
+
+cmake_build() {
+    cmake --build $DYNAFLOW_LAUNCHER_BUILD_DIR --target install -j $DYNAFLOW_LAUNCHER_PROCESSORS_USED
```

Comment:
I don't know if this syntax works with the minimum cmake required (maybe the same in dynawo...)

Reply:
It works

Reply:
works in 3.12, which is the mnimum required for DFL

Reply:
ok